WasatchNET/SPISpectrometer.cs

```diff
@@ -396,11 +685,143 @@ public override LaserPowerResolution laserPowerResolution
+
+                */
+
+                byte[] transmitData = new byte[0];
+
+                transmitData = wrapCommand(CCD_GAIN, transmitData, 40);
+
+                byte[] result = padding(10);
+                    
+                result = spi.readWrite(transmitData);
+
+                byte[] final = new byte[2];//{ result[0], result[1] };
+
+                int index = 0;
+
+                while (index < result.Length)
+                {
+                    if (result[index] == 0x3c)
+                        break;
+                    ++index;
+                }
+
+                while (result[index] == 0x3c)
+                    ++index;
+                --index;
+
+                final[1] = result[index + 5];
+                final[0] = result[index + 4]; 
+
+                //byte[] result = new byte[2];
+
+                return FunkyFloat.toFloat(Unpack.toUshort(final));
+
+
+            }
+            set
+            {
+                ushort funkyTransform = FunkyFloat.fromFloat(value);
+                byte[] transmitData = new byte[2];
+
+                transmitData[1] = (byte)(((uint)funkyTransform & 0xFF00) >> 8);
+                transmitData[0] = (byte)((uint)funkyTransform & 0x00FF);
+
+                byte[] command = wrapCommand(SET_CCD_GAIN, transmitData, 40);
+
+                byte[] result = spi.readWrite(command);
+
+                //eadOnce.Add(Opcodes.GET_DETECTOR_GAIN);
+                //sendCmd(Opcodes.SET_DETECTOR_GAIN, FunkyFloat.fromFloat(detectorGain_ = value));
+            }
+        }
 
         public override float detectorGainOdd { get => 0; }
 
-        public override short detectorOffset { get => 0; }
+        public override short detectorOffset 
+        {
+            get
+            {
+                // if (kludgedOut) return 0;
+                /*
+                const Opcodes op = Opcodes.GET_DETECTOR_GAIN;
+                if (haveCache(op))
+                    return detectorGain_;
+                readOnce.Add(op);
+                return detectorGain_ = FunkyFloat.toFloat(Unpack.toUshort(getCmd(op, 2)));
+                */
+
+                byte[] transmitData = new byte[0];
+
+                transmitData = wrapCommand(CCD_OFFSET, transmitData, 40);
+
+                byte[] result = padding(10);
+
+                result = spi.readWrite(transmitData);
+
+                byte[] final = new byte[2];//{ result[0], result[1] };
+
+                int index = 0;
+
+                while (index < result.Length)
+                {
+                    if (result[index] == 0x3c)
+                        break;
+                    ++index;
+                }
+
+                while (result[index] == 0x3c)
+                    ++index;
+                --index;
+
+                final[1] = result[index + 5];
+                final[0] = result[index + 4];
+
+
+
+                //byte[] result = new byte[2];
+
+                short wrongForm = Unpack.toShort(final);
```

Comment:
I REALLY hope we end up with a proper Int16 before this gets locked into any long-term API.  This is stupid.